Example/AblyChatExample/Mocks/Mocks.swift

```diff
@@ -0,0 +1,190 @@
+import Ably
+import AblyChat
+
+final class MockChatClient: ChatClient {
+    
+    static let shared = MockChatClient(
+        realtime: MockRealtime(key: ""),
+        clientOptions: ClientOptions()
+    )
+    
+    public init(realtime _: ARTRealtimeProtocol, clientOptions _: ClientOptions?) {
+        // This one doesn’t do `fatalError`, so that I can call it in the example app
+    }
+
+    public var rooms: any Rooms {
+        MockRooms()
+    }
+
+    public var connection: any Connection {
+        fatalError("Not yet implemented")
+    }
+
+    public var clientID: String {
+        fatalError("Not yet implemented")
+    }
+
+    public var realtime: any ARTRealtimeProtocol {
+        fatalError("Not yet implemented")
+    }
+
+    public var clientOptions: ClientOptions {
+        fatalError("Not yet implemented")
+    }
+}
+
+final class MockRooms: Rooms {
+    
+    func get(roomID: String, options: RoomOptions) -> any Room {
+        MockRoom(roomID: "Demo Room")
+    }
+    
+    func release(roomID: String) async throws {
+        fatalError("Not yet implemented")
+    }
+    
+    var clientOptions: ClientOptions {
+        fatalError("Not yet implemented")
+    }
+}
+
+final class MockRoom: Room {
+    
+    let roomID: String
+    
+    init(roomID: String) {
+        self.roomID = roomID
+    }
+    
+    var messages: any Messages {
+        MockMessages(clientID: "Demo", roomID: roomID)
+    }
+
+    var presence: any Presence {
+        fatalError("Not yet implemented")
+    }
+
+    var reactions: any RoomReactions {
+        fatalError("Not yet implemented")
+    }
+
+    var typing: any Typing {
+        fatalError("Not yet implemented")
+    }
+
+    var occupancy: any Occupancy {
+        fatalError("Not yet implemented")
+    }
+    
+    var status: any RoomStatus {
+        fatalError("Not yet implemented")
+    }
+    
+    func attach() async throws {
+        fatalError("Not yet implemented")
+    }
+    
+    func detach() async throws {
+        fatalError("Not yet implemented")
+    }
+    
+    var options: RoomOptions {
+        fatalError("Not yet implemented")
+    }
+}
+
+final class MockMessages: @unchecked Sendable, Messages, ObservableObject {
+    let clientID: String
+    let roomID: String
+    
+    @Published var log = [Message]()
```

Comment:
What's the purpose of this and the `ObservableObject` conformance? It doesn't seem related to the SDK’s API.

Reply:
See https://github.com/ably-labs/ably-chat-swift/pull/16#issuecomment-2303070970